code/workspaces/client-api/src/schema/resolvers.js

```diff
@@ -0,0 +1,82 @@
+    datalab: (obj, { name }, { user }) => datalabRepository.getByName(user, name),
+    datalabs: (obj, args, { user }) => datalabRepository.getAll(user),
+    userPermissions: (obj, params, { token }) => getUserPermissions(token),
+    checkNameUniqueness: (obj, { name }, { user, token }) => permissionChecker([STACKS_CREATE, STORAGE_CREATE], user, () => internalNameChecker({ user, token }, name)),
+    users: (obj, args, { user, token }) => permissionChecker(USERS_LIST, user, () => userService.getAll({ token })),
+    project: (obj, { id }, { user }) => permissionChecker(SETTINGS_READ, user, () => projectService.getProjectById(id)),
+  },
+
+  Mutation: {
+    createStack: (obj, { stack }, { user, token }) => permissionChecker(STACKS_CREATE, user, () => stackApi.createStack({ user, token }, DATALAB_NAME, stack)),
+    deleteStack: (obj, { stack }, { user, token }) => permissionChecker(STACKS_DELETE, user, () => stackApi.deleteStack({ user, token }, DATALAB_NAME, stack)),
+    createDataStore: (obj, { dataStore }, { user, token }) => permissionChecker(STORAGE_CREATE, user, () => dataStoreApi.createDataStore({ user, token }, DATALAB_NAME, dataStore)),
+    deleteDataStore: (obj, { dataStore }, { user, token }) => permissionChecker(STORAGE_DELETE, user, () => dataStoreApi.deleteDataStore({ user, token }, DATALAB_NAME, dataStore)),
+    addUserToDataStore: (obj, { dataStore: { name, users } }, { user }) => permissionChecker(STORAGE_EDIT, user, () => dataStorageRepository.addUsers(user, name, users)),
+    removeUserFromDataStore: (obj, { dataStore: { name, users } }, { user }) => permissionChecker(STORAGE_EDIT, user, () => dataStorageRepository.removeUsers(user, name, users)),
+    addProjectPermission: (obj, { permission: { projectId, userId, role } }, { user }) => (
+      permissionChecker(PERMISSIONS_CREATE, user, () => projectService.addProjectPermission(projectId, userId, role))
+    ),
+    removeProjectPermission: (obj, { permission: { projectId, userId, role } }, { user }) => (
+      permissionChecker(PERMISSIONS_DELETE, user, () => projectService.removeProjectPermission(projectId, userId, role))
+    ),
+  },
+
+  DataStore: {
+    users: (obj, args, { user }) => permissionChecker(USERS_LIST, user, () => obj.users),
+    accessKey: (obj, args, { user }) => minioTokenService.requestMinioToken(obj, user),
+    stacksMountingStore: ({ name }, args, { user, token }) => stackService.getAllByVolumeMount({ user, token }, name),
+    status: () => READY,
+  },
+
+  Stack: {
+    redirectUrl: (obj, args, { user }) => stackUrlService(obj, user),
+  },
+
+  Project: {
+    projectUsers: (obj, args, ctx) => userService.getProjectUsers(args.projectId, ctx.token),
+  },
+
+  ProjectUser: {
+    name: (obj, args, ctx) => userService.getUserName(obj, ctx.token),
+  },
+
+  // This mapping is required to map the string to an id in the database.
```

Comment:
Has this been introduced in this update? Wondering if we are in the position to fix this at the moment.

Reply:
Yes, I was tempted but realised it would break the existing CEH instance so would need to be carefully coordinated. Will add a ticket to the backlog to fix it.